### PR TITLE
Fix test packages ambiguous `throws`

### PR DESCRIPTION
**Pull request by desistefanova**
Hides throws from 'package:test/expect.dart'.
Critical for main branch.

Disable test "Realm path with unicode symbols" for Android and IOS.
Fix job status in case of error while creating the App services

### CODE REVIEW

File: test/realm_test.dart [hunk 1]
```diff
@@ -1864,7 +1864,7 @@ Future<void> main([List<String>? args]) async {
     var config = Configuration.local([Car.schema], path: "${generateRandomUnicodeString()}.realm");
     var realm = getRealm(config);
     expect(realm.isClosed, false);
-  });
+  }, skip: Platform.isAndroid || Platform.isIOS);
```
Comment (nirinchev) [inline test/realm_test.dart:1867]: Why is this needed?

Comment (nielsenko) [reply test/realm_test.dart:1867]: [test_api-0.5.0 ](https://pub.dev/packages/test_api) was released 3 days ago.

Comment (blagoev) [reply test/realm_test.dart:1867]: Shold we create a task to investigate and remove this later?

Comment (blagoev) [reply test/realm_test.dart:1867]: Also if it fails on Android why disable for iOS?

Comment (desistefanova) [reply test/realm_test.dart]: The tests fail for both [Android](https://github.com/realm/realm-dart/actions/runs/4531527657/jobs/7982684657#step:9:1215) and [IOS ](https://github.com/realm/realm-dart/actions/runs/4531527657/jobs/7982684404#step:9:2484) when having unicode symbols in the realm path. I will disable these tests in order to be able to fix the main branch. I [created a new issue for the paths.](https://github.com/realm/realm-dart/issues/1230)

File: test/test.dart [hunk 2]
```diff
@@ -532,26 +532,30 @@ extension on Map<String, String?> {
...
+    client.publicRSAKey = publicRSAKeyForJWTValidation;
 
-  var apps = await client.getOrCreateApps();
-  baasApps.addAll(apps);
-  _baasClient = client;
+    var apps = await client.getOrCreateApps();
+    baasApps.addAll(apps);
+    _baasClient = client;
+  } catch (e) {
+    baasApps.clear();
```
Comment (blagoev) [inline test/test.dart:557]: we are silencing all exceptions here. Not sure why we are doing that. Then it seems we allow the run to continue. What is the intention here?

Comment (desistefanova) [reply test/test.dart:557]: Right, I would've printed the exception. Yes, The tests continue to be run and all the local tests would pass and the sync tests will fail. This will cause the job to fail based on the test report. Otherwise, the tests are not ran and the job doesn't register any errors.

File: lib/src/cli/atlas_apps/baas_client.dart [hunk 3]
```diff
@@ -449,14 +449,14 @@ class BaasClient {
...
     while (true) {
       try {
-        await _patch('groups/$_groupId/apps/$app/services/$mongoServiceId/config', syncConfig);
         await _post('groups/$_groupId/apps/$app/services/$mongoServiceId/default_rule', rules);
+        await _patch('groups/$_groupId/apps/$app/services/$mongoServiceId/config', syncConfig);
         break;
       } catch (err) {
-        if (attempt++ < 24) {
-          print('Failed to update service after ${attempt * 5} seconds. Will keep retrying ...');
+        if (attempt++ < 30) {
```
[the remark this answers is not in the document]
Comment (desistefanova) [reply lib/src/cli/atlas_apps/baas_client.dart]: This change, didn't help us. We will investigate the issue with creating the service.

File: lib/src/cli/atlas_apps/baas_client.dart [hunk 4]
```diff
@@ -357,12 +358,17 @@ class BaasClient {
...
-    final dynamic createUserResult = await _post('groups/$_groupId/apps/$appId/users', '{"email": "realm-test@realm.io", "password":"123456"}');
-    print("Create user result: $createUserResult");
+      );
+      await _put('groups/$_groupId/apps/$appId/sync/config', '{ "development_mode_enabled": true }');
+
+      //create email/password user for tests
+      final dynamic createUserResult = await _post('groups/$_groupId/apps/$appId/users', '{"email": "realm-test@realm.io", "password":"123456"}');
+      print("Create user result: $createUserResult");
+    } catch (error) {
+      print(error);
```
Comment (blagoev) [inline lib/src/cli/atlas_apps/baas_client.dart:368]: Do we need to print the error here?

File: test/test.dart [hunk 5]
```diff
@@ -532,26 +532,31 @@ extension on Map<String, String?> {
...
+        ? BaasClient.docker(baasUrl, differentiator)
+        : BaasClient.atlas(baasUrl, cluster, apiKey!, privateApiKey!, projectId!, differentiator));
+
+    client.publicRSAKey = publicRSAKeyForJWTValidation;
+
+    var apps = await client.getOrCreateApps();
+    baasApps.addAll(apps);
+    _baasClient = client;
+  } catch (error) {
+    print(error);
```
Comment (blagoev) [inline test/test.dart:557]: same as above

File: test/test.dart [hunk 6]
```diff
@@ -561,6 +566,7 @@ Future<void> baasTest(
   AppNames appName = AppNames.flexible,
   dynamic skip,
 }) async {
+  if (BaasClient.initializationError != null) throw BaasClient.initializationError!;
```
Comment (blagoev) [inline test/test.dart:569]: small one but... can we wrap this in brackets?
```suggestion
  if (BaasClient.initializationError != null){
   throw BaasClient.initializationError!;
   }
```

File: test/test.dart [hunk 7]
```diff
@@ -581,6 +587,7 @@ Future<AppConfiguration> getAppConfig({AppNames appName = AppNames.flexible}) as
 
   final app = baasApps[appName.name] ??
       baasApps.values.firstWhere((element) => element.name == BaasClient.defaultAppName, orElse: () => throw RealmError("No BAAS apps"));
+  if (app.error != null) throw app.error!;
```
Comment (blagoev) [inline test/test.dart:590]: same as above nit

File: test/test.dart [hunk 8]
```diff
@@ -532,26 +532,31 @@ extension on Map<String, String?> {
 
 BaasClient? _baasClient;
 Future<void> setupBaas() async {
-  final baasUrl = arguments[argBaasUrl];
-  if (baasUrl == null) {
-    return;
+  if (BaasClient.initializationError != null) return;
```
Comment (blagoev) [inline test/test.dart:535]: why do we need this as static on the `BaasClient`? Can we move it to module level variable?

**Review by blagoev — CHANGES_REQUESTED** [review]

File: lib/src/cli/atlas_apps/baas_client.dart (hunk 4 above)
Comment (desistefanova) [reply lib/src/cli/atlas_apps/baas_client.dart:368]: Yes, I would like to see which of the actions of app services creation are passed successfully before the error.

File: test/test.dart (hunk 8 above)
Comment (desistefanova) [reply test/test.dart]: Done.

File: lib/src/cli/atlas_apps/baas_client.dart (hunk 4 above)
Comment (blagoev) [reply lib/src/cli/atlas_apps/baas_client.dart:367]: I agree we can leave it for the test.dart but here I think we should not `print`.
If you want to see which apps succeeded you can print them in the tests.dart

**Review by blagoev — APPROVED** [review]

**Review by nielsenko — APPROVED** [review]
